File: Exercise_2.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int a1 =m-1;
        int a2 =n-1;
        int in =m+n-1;
        while(a1>=0 && a2>=0){
            if(nums1[a1] > nums2[a2]){
                nums1[in] = nums1[a1];
                --a1;
            } else {
                nums1[in] = nums2[a2];
                --a2;
            }
            --in;
        }
        while(a2>=0){
            nums1[in] = nums2[a2];
                --a2;
                --in;
        }
    }
}

//tc=O(m+n)
//sc=O(1)

File: Exercise_2.cpp
// Time Complexity                              :  O(N)
// Space Complexity                             :  O(1)
// Did this code successfully run on Leetcode   :  Yes
// Any problem you faced while coding this      :  No

#include <bits/stdc++.h> 
using namespace std;  

// https://leetcode.com/problems/merge-sorted-array/
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        
        vector<int> ans(m+n);
        int mn = min(m,n);
        
        int r1 = m - 1;
        int r2 = n - 1;
        int r3 = m + n - 1;
        // using three pointer technique to merge 2 sorted arrays
        while(r2 >= 0 and r1 >= 0) {
            if(nums1[r1] > nums2[r2]) {
                nums1[r3] = nums1[r1];
                r1--;
            } else {
                nums1[r3] = nums2[r2];
                r2--;
            }
            r3--;
        }
        
        while(r2 >= 0) {
            nums1[r3] = nums2[r2];
            r2--;
            r3--;
        }
        
    }
};